src/Illuminate/Support/Number.php

```diff
@@ -287,14 +297,26 @@ public static function withLocale(string $locale, callable $callback)
     /**
      * Set the default locale.
      *
-     * @param  string  $locale
+     * @param  callable|string  $locale
      * @return void
      */
-    public static function useLocale(string $locale)
+    public static function useLocale(callable|string $locale)
     {
         static::$locale = $locale;
     }
 
+    /**
+     * Check $setLocale and set the locale if it is not null.
+     *
+     * @return string
+     */
+    protected static function checkLocale()
```

Comment:
Suggestion:

1. Rename this method to `locale()`
2. Cache the first call if it is a callable

```php
protected static function checkLocale()
{
    if (is_callable(static::$locale)) {
        static::$locale = call_user_func(static::$locale); 
    }

    return static::$locale;
}
```

Note that checking if the callable is not a string is wrong, as PHP allows callable strings, by using a function name, including a class' static method.

Reference: https://www.php.net/manual/en/language.types.callable.php

For example:

```php
<?php

namespace App {
    function getLocale() {
        return 'pt';
    }

    class Foo
    {
        public static function locale()
        {
            return 'es';
        }
    }
}

namespace {
    var_dump(
        \is_callable('trim'),
        \is_callable('App\\getLocale'),
        \is_callable('App\\Foo::locale'),
    );
}
```

Outputs:

```
bool(true)
bool(true)
bool(true)
```

https://3v4l.org/McEZr